nemo/collections/asr/models/classification_models.py

```diff
@@ -40,7 +40,7 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self.preprocessor = EncDecClassificationModel.from_config_dict(self._cfg.preprocessor)
         self.encoder = EncDecClassificationModel.from_config_dict(self._cfg.encoder)
         self.decoder = EncDecClassificationModel.from_config_dict(self._cfg.decoder)
-        self.loss = CrossEntropyLoss(logits_ndim=self.decoder.num_classes)
+        self.loss = CrossEntropyLoss()
```

Comment:
So how come you removed this from asr models but had to add it for the nlp models?

Reply:
The CrossEntropyLoss has the default value of `logits_ndim=2`. basically a flattened matrix shape of [batch, dimension] is mapped to this loss. This is the speech commands file, not asr, so that is the appropriate logits shape.

For NLP, they perform per timestep token generation, so logits are of shape [batch, timestep, dimension], and therefore I need to explicitly override the `logits_ndim=3` part for NLP. This has been verified by NLP team.

Reply:
so ndim is shape rank, not number of classes, right?

Reply:
Yup thats right. I misunderstood the ndim part when I first wrote this line.